proposal/first-class-mixins.md

```diff
@@ -0,0 +1,199 @@
+>
+> As an example, if we declare two mixins:
+> 
+> ```scss
+> @mixin mixin1 {
+>   color: red;
+> }
+> 
+> $a: meta.get-mixin(mixin1);
+> 
+> @mixin mixin1 {
+>   color: red;
+> }
+> 
+> $b: meta.get-mixin(mixin1);
+> ```
+> 
+> Although every aspect of the two mixins is the same, `$a != $b`, because they refer to separate mixin declarations/objects.
+
+### Serialization
+
+To serialize a `Mixin`:
+
+* If the value is not being inspected, throw an error.
+
+* Otherwise:
+
+  * Emit `"get-mixin("`.
+
+  * Emit a double quote (`"`), then the `name` of the mixin, then another double quote.
+
+  * Emit `")"`.
+
+## Functions
+
+### `meta.type-of()`
+
+Add the following clause to the [`meta.type-of()`] function and the top-level
+`type-of()` function:
+
+[`meta.type-of()`]: ../spec/built-in-modules/meta.md#type-of
+
+* If `$value` is a mixin, return an unquoted string with value `"mixin"`.
+
+### `meta.get-mixin()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.get-mixin($name, $module: null)
+```
+
+* If `$name` is not a string, throw an error.
+
+* If `$module` is null:
+
+  * Return the result of resolving a mixin named `$name`. If this returns
+    null, throw an error.
+
+* Otherwise:
+  
+  * If `$module` is not a string, throw an error.
+
+  * Let `use` be the `@use` rule in [the current source file][] whose
+    namespace is equal to `$module`. If no such rule exists, throw an error.
+
+  * Return [`use`'s module][]'s mixin named `$name`, or throw an error if no
+    such mixin exists.
+
+  [the current source file]: ../spec.md#current-source-file
+  [`use`'s module]: at-rules/use.md#a-use-rules-module
+
+### `meta.module-mixins()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.module-mixins($module)
+```
+
+* If `$module` is not a string, throw an error.
+
+* Let `use` be the `@use` rule in [the current source file][] whose namespace is
+  equal to `$module`. If no such rule exists, throw an error.
+
+* Return a map whose keys are the quoted string names of mixins in [`use`'s module][] and
+  whose values are the corresponding mixins.
+
+### `meta.accepts-content()`
+
+This is a new function in the `sass:meta` module.
+
+```
+meta.accepts-content($mixin)
+```
+
+* If `$mixin` is not a mixin, throw an error.
+
+* Return a boolean which is true if the body of the mixin has an `@content` rule. This is the same value as `mixin.hasContent`.
+
+## Mixins
+
+### `meta.apply()`
+
+```
+meta.apply($mixin, $args...)
+```
+
+> Mixins must be invoked with `@include`, so `include` is guaranted to exist.
+
+* If `$mixin` is not a mixin, throw an error.
+
+* If `include` has a `ContentBlock`, it should be passed down to `$mixin`.
+
+* Assign $args to `$mixin`'s ArgumentDeclaration in `$mixin`'s scope.
```

Comment:
Elsewhere we use the phrasing "execute an `ArgumentInvocation` with an `ArgumentDeclaration`". We don't have an `ArgumentInvocation` (since the args are already resolved by the time this is running), but you can create one and say "execute the `ArgumentInvocation` `(...$args)` with `$mixin`'s `ArgumentDeclaration` in `$mixin`'s scope."

(This line is also missing a few code quotes.)